WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverAdapter.kt

```diff
@@ -48,7 +48,9 @@ class ReaderDiscoverAdapter : Adapter<ReaderViewHolder>() {
                 return false
             }
             return when (oldItem) {
-                is ReaderPostUiState -> oldItem.id == (newItem as ReaderPostUiState).id
+                is ReaderPostUiState -> {
+                    oldItem.postId == (newItem as ReaderPostUiState).postId && oldItem.blogId == newItem.blogId
```

Comment:
This change is unrelated, but I realized my mistake. PostId in Reader isn't unique as there might be posts from different sites.